packages/cli/src/gemini.tsx

```diff
@@ -304,6 +303,23 @@ export async function main() {
 
   const argv = await parseArguments(settings.merged);
 
+  if (
```

Comment:
where is extension command processing happening before this change? point is to not have a separate code path for it if you can avoid it. alternately if you can point me to where it is happening you can just create a shared helper for the initialization done early in runNonInteractive that makes stdout and stderr function normally now that we know the user is not interactive.